AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/Resolvers/Authentication/AuthenticationState+Resolver.swift

```diff
@@ -19,15 +19,23 @@ extension AuthenticationState {
         ) -> StateResolution<StateType> {
             switch oldState {
             case .notConfigured:
-                guard let authEvent = event as? AuthenticationEvent else {
+                if let authEvent = event as? AuthenticationEvent {
+                    return resolveNotConfigured(byApplying: authEvent)
+                } else if let authZEvent = event.isAuthorizationEvent,
+                        case .startFederationToIdentityPool = authZEvent {
+                    return .init(newState: .federatingToIdentityPool)
+                } else {
                     return .from(oldState)
                 }
-                return resolveNotConfigured(byApplying: authEvent)
             case .configured:
-                guard let authEvent = event as? AuthenticationEvent else {
+                if let authEvent = event as? AuthenticationEvent {
```

Comment:
Even configured is an intermediate state, it will move to signedOut after this, should we handle `startFederationToIdentityPool` inside signedOut?